File: qna/src/main/java/com/example/rag/qna/adapter/output/question/QuestionPersistencePort.java
package com.example.rag.qna.adapter.output.question;

import com.example.rag.qna.domain.Question;

import java.util.List;

public interface QuestionPersistencePort {
    void saveQuestion(Question question);
    void updateQuestion(Question question);
    void deleteQuestion(String questionId);
    Question findQuestionById(String questionId);
    List<Question> findQuestionsByUserId(String userId, int offset, int limit);
}

File: qna/src/main/kotlin/com/example/rag/qna/adapter/output/question/QuestionPersistencePort.kt
package com.example.rag.qna.adapter.output.question

import com.example.rag.qna.domain.Question

interface QuestionPersistencePort {
    fun saveQuestion(domain: Question): Question
    fun updateQuestion(domain: Question)
    fun deleteQuestion(questionId: String)
    fun findQuestionById(questionId: String): Question
    fun findQuestionsByUserId(userId: String, offset: Int, limit: Int): List<Question>
}